app/services/task_service/gpt_generate_tests.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+module TaskService
+  class GptGenerateTests < ServiceBase
+    def initialize(task:)
+      super()
+      @task = task
+      @message = task.description
+      @language = @task.programming_language.language || 'Python'
+    end
+
+    def execute
+      file_content = gpt_response
+      test_file = TaskFile.new(content: file_content, name: 'test.py', used_by_grader: true, visible: false, xml_id: SecureRandom.uuid)
+      testing_framework = TestingFramework.find_or_initialize_by(name: 'Pytest', version: 6)
+      test = Test.new(task: @task, title: 'AI-generated Test', xml_id: SecureRandom.uuid, files: [test_file], testing_framework:)
+      @task.tests << test
+      @task.save!
```

Comment:
The first line (`@task.tests << test`) will probably save the task already; do we need the second one?

Reply:
From a quick search, it seems that both commands perform slightly different things therefore both are required. at least that's what I understood.

Reply:
Sure, usually they do, but not necessarily in this example. However, it is unexpected that line 17 already saves the task as well, this is what I meant. Feel free to use a debugger to observe the behavior and place a breakpoint before executing line 18.

Reply:
Yes, I outcommented it and everything still worked fine. Will remove this line. Thanks

Reply:
What do you suggest I write instead?

Reply:
Ah, don't mind. You are not changing the task at all, so that this method is fine.